### PR TITLE
web3: added option to enable/disable autoconnect

### LINKED ISSUE #4957: web3: include an option to enable/disable autoconnect
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Request

Currently web3 supports a feature of `reconnect` by which we can configure the reconnection parameters. A similar feature is needed to enable/disable the autoconnect to the mentioned provider.

Specifically, enveloping the condition of connection only if autoconnect parameter is set as true [here](https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-providers-ws/src/index.js#L89) and similar places.

### Use Cases

There are certain scenarios where this feature can come in handy, like when we want to test some codebase containing web3 library, but cannot pass the provider; in such case we can simple turn off autoconnect to avoid connection in the first place.

One of such usages can come in handy while testing [this](https://github.com/hyperledger/cactus/blob/dda3f00c5319e22b03c73fe6dbc645dea41cc58e/packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts#L151)

### Implementation Ideas

This can be easily implemented by enveloping [this](https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-providers-ws/src/index.js#L89) with a condition which checks if the `autoconnect` parameter is set or not (should be defaulted to `true` to avoid possible code crashes for devs upgrading to the version of web3 including this feature)

### Feature Examples/References

_No response_

### Are you willing to implement this feature?

- [X] Yes

### PR DESCRIPTION
**Pull request by TheJagpreet**
## Description
Added a boolean to enable/disable autoconnect

Fixes #4957

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: packages/web3-providers-ws/src/index.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ export default class WebSocketProvider<
 		clientUrl: string,
 		wsProviderOptions?: ClientOptions | ClientRequestArgs,
 		reconnectOptions?: ReconnectOptions,
+		autoConnect?: boolean,
```
Comment (jdevcs) [inline packages/web3-providers-ws/src/index.ts:74]: There is already [autoConnect](https://github.com/ChainSafe/web3.js/blob/82574a2587810f9bea26ba65da48fe667162e573/packages/web3-providers-ws/src/types.ts#L21) in `reconnectOptions` 

```suggestion
		
```

**Review by jdevcs — CHANGES_REQUESTED** [review]

File: packages/web3-providers-ws/src/index.ts [hunk 2]
```diff
@@ -98,7 +99,9 @@ export default class WebSocketProvider<
 		this._onCloseHandler = this._onClose.bind(this);
 
 		this._init();
-		this.connect();
+		if(autoConnect!==false){
+			this.connect();
```
Comment (jdevcs) [inline packages/web3-providers-ws/src/index.ts:103]: This is for first time connection , not for re connect so what is need of adding this check here?

File: packages/web3-providers-ws/src/index.ts [hunk 3]
```diff
@@ -185,6 +188,7 @@ export default class WebSocketProvider<
 		Method extends Web3APIMethod<API>,
 		ResponseType = Web3APIReturnType<API, Method>,
 	>(request: Web3APIPayload<API, Method>): Promise<JsonRpcResponse<ResponseType>> {
+		this.connect();
```
Comment (jdevcs) [inline packages/web3-providers-ws/src/index.ts:191]: Validate will be called before init so why this connect is added here?

File: packages/web3-providers-ws/src/index.ts (hunk 1 above)
Comment (TheJagpreet) [reply packages/web3-providers-ws/src/index.ts:74]: Oh, maybe I wasn't using the version of web3 which has this.
Closing the PR as the feature already exists,

**Pull request closed without merging**